lib/inspec/resources/sys_info.rb

```diff
@@ -34,15 +52,46 @@ def hostname(opt = nil)
       elsif os.darwin?
         mac_hostname(opt)
       elsif os.windows?
-        if !opt.nil?
-          skip_resource "The `sys_info.hostname` resource is not supported with that option on your OS."
-        else
-          inspec.powershell("$env:computername").stdout.chomp
-        end
+        opt = "s" if opt.nil?
+        windows_hostname(opt)
       else
         skip_resource "The `sys_info.hostname` resource is not supported on your OS yet."
       end
     end
+    
+    def windows_hostname(opt = nil)
+      if @cache.nil?
+        @cache = win_cache
+      end
+      if !opt.nil?
```

Comment:
`unless opt.nil?` is nicer than the negation. But since we're just checking that opt is set, `if opt` should suffice and the `else` can cover the case where its nil.